app/models/smart_proxy.rb

```diff
@@ -107,6 +108,44 @@ def version
     result
   end
 
+  def plugin_version(plugin_name)
+    raise ::Foreman::Exception.new(N_('Plugin name cannot be blank')) if plugin_name.blank?
+    result = {}
+    begin
+      Timeout::timeout(20) do
+        version = ProxyAPI::Version.new(:url => url).plugin_versions
+        result[:success] = true
+        result[:message] = version[plugin_name]
+      end
+    rescue *HTTP_ERRORS => exception
+      raise ::Foreman::WrappedException.new exception, N_("Unable to connect to smart proxy")
+    end
+    Rails.cache.write("plugin_#{plugin_name}_proxy_#{self.id}", result, {:expires_in  => 5.minutes })
```

Comment:
You write to cache, but never read it?

Also, I think it's much more robust to put caching into the ProxyAPI::* classes than here. We can simply cache each HTTP response. This could be generic method in ProxyAPI: cached_get instead of get. URL and parameters can be formatted as key: "proxy_cached_get_ #{url}_#{params}" in some way.

Expires in should be a configurable option.